test/chai-as-promised/chai-as-promised.js

```diff
@@ -0,0 +1,36 @@
+/**
+ * Copyright 2017 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ *
+ * NOTE: This special file is required for three reasons:
+ *
+ * 1. The AMP unit tests use the 'chai' v4 assertion library, which depends on
+ *    'chai-as-promised' v7, a library that contains ES6 code.
+ *    See https://github.com/domenic/chai-as-promised/issues/133.
+ *
+ * 2. We cannot use 'chai-as-promised' on Sauce labs, since some browsers only
+ *    support ES5 code.
+ *    See https://github.com/domenic/chai-as-promised#browsernode-compatibility.
+ *
+ * 3. Transpiling 'chai-as-promised' with babelify is not possible
+ *    since directories under `node_modules/` are ignored by default.
+ *    See https://github.com/babel/babelify/issues/53.
+ *
+ * We therefore include this file while running unit tests (only run on chrome),
+ * but not while running integration tests (run on multiple browsers).
+ * As a result, integration tests may not use `chai-as-promised.
+ */
+
+chai.use(require('chai-as-promised'));
```

Comment:
Good sleuthing. Let's discuss perhaps deprecating and removing `chai-as-promised`.

Reply:
For the record, I found hundreds of places where `chai-as-promised` was being used in the unit tests, while there were only 2 (unnecessary) uses of the library in the integration tests.

In general, I think it's a good idea to retain the use of `chai-as-promised` in the unit tests, since otherwise, a 5-10 line snippet of verification code can balloon into dozens of lines, that are harder to write and easier to screw up. IMO, using `chai-as-promised` makes the intention of the test case writer a lot clearer than cascading `.then` blocks would do.

Also, the current solution of restricting the use of this library to the unit tests works well from the initial exploring I've done, and I think it's sustainable in the medium term.

Happy to discuss this in a separate Issue / PR.